examples/objective-c/hello_world/ViewController.m

```diff
@@ -65,38 +65,38 @@ - (void)performRequest {
   self.requestCount++;
   NSLog(@"Starting request to '%@'", _REQUEST_PATH);
 
-  int requestID = self.requestCount;
   // Note: this request will use an http/1.1 stream for the upstream request.
   // The Swift example uses h2. This is done on purpose to test both paths in end-to-end tests
   // in CI.
-  RequestBuilder *builder = [[RequestBuilder alloc] initWithMethod:RequestMethodGet
-                                                            scheme:_REQUEST_SCHEME
-                                                         authority:_REQUEST_AUTHORITY
-                                                              path:_REQUEST_PATH];
-  Request *request = [builder build];
-  ResponseHandler *handler = [[ResponseHandler alloc] initWithQueue:dispatch_get_main_queue()];
+  int requestID = self.requestCount;
+  RequestHeadersBuilder *builder = [[RequestHeadersBuilder alloc] initWithMethod:RequestMethodGet
+                                                                          scheme:_REQUEST_SCHEME
+                                                                       authority:_REQUEST_AUTHORITY
+                                                                            path:_REQUEST_PATH];
+  [builder addUpstreamHttpProtocol:UpstreamHttpProtocolHttp1];
+  RequestHeaders *headers = [builder build];
 
   __weak ViewController *weakSelf = self;
-  [handler onHeaders:^(NSDictionary<NSString *, NSArray<NSString *> *> *headers,
-                       NSInteger statusCode, BOOL endStream) {
-    NSLog(@"Response status (%i): %ld\n%@", requestID, statusCode, headers);
-    NSString *body = [NSString stringWithFormat:@"Status: %ld", statusCode];
+  StreamPrototype *prototype = [self.client newStream];
```

Comment:
I'd be fine with `newStream` or `newStreamPrototype`, but if you're going with the latter, looks like you missed this.